jtlibrary/python/jtmodules/src/jtmodules/segment_secondary_cp.py

```diff
@@ -0,0 +1,338 @@
+#         this list of conditions and the following disclaimer.
+#
+#     2.  Redistributions in binary form must reproduce the above copyright
+#         notice, this list of conditions and the following disclaimer in the
+#         documentation and/or other materials provided with the distribution.
+#
+#     3.  Neither the name of the Broad Institute, Inc. nor the names of its
+#         contributors may be used to endorse or promote products derived from
+#         this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED "AS IS."  BROAD MAKES NO EXPRESS OR IMPLIED
+# REPRESENTATIONS OR WARRANTIES OF ANY KIND REGARDING THE SOFTWARE AND
+# COPYRIGHT, INCLUDING, BUT NOT LIMITED TO, WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE, CONFORMITY WITH ANY DOCUMENTATION,
+# NON-INFRINGEMENT, OR THE ABSENCE OF LATENT OR OTHER DEFECTS, WHETHER OR NOT
+# DISCOVERABLE. IN NO EVENT SHALL BROAD, THE COPYRIGHT HOLDERS, OR CONTRIBUTORS
+# BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO PROCUREMENT OF SUBSTITUTE
+# GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+# OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF, HAVE REASON TO KNOW, OR IN
+# FACT SHALL KNOW OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# If, by operation of law or otherwise, any of the aforementioned warranty
+# disclaimers are determined inapplicable, your sole remedy, regardless of the
+# form of action, including, but not limited to, negligence and strict
+# liability, shall be replacement of the software with an updated version if one
+# exists.
+#
+# Development of CellProfiler has been funded in whole or in part with federal
+# funds from the National Institutes of Health, the National Science Foundation,
+# and the Human Frontier Science Program.
+
+'''Jterator module for segmentation of secondary objects around existing
+primary objects.
+'''
+
+import logging
+import numpy as np
+import mahotas as mh
+import collections
+import centrosome.cpmorphology
+import centrosome.propagate
+import scipy.ndimage
+import skimage.morphology
+
+logger = logging.getLogger(__name__)
+
+VERSION = '0.0.1'
+M_PROPAGATION = "propagation"
+M_WATERSHED_G = "watershed_gradient"
+M_WATERSHED_I = "watershed_image"
+M_DISTANCE_N = "distance_n"
+M_DISTANCE_B = "distance_b"
+
+Output = collections.namedtuple('Output', ['secondary_label_image', 'figure'])
+
+
+def filter_labels(primary_label_image, secondary_label_image):
+
+    labels_in = np.unique(primary_label_image) - [0]
+    labels_out = np.unique(secondary_label_image) - [0]
+
+    extra_labels = set(labels_out) - set(labels_in)
+    missing_labels = set(labels_in) - set(labels_out)
+
+    if len(extra_labels) != 0:
```

Comment:
```suggestion
    if extra_labels:
```
Standard data structures (such as sets), support boolean evaluation.
https://docs.python.org/3/library/stdtypes.html#truth-value-testing